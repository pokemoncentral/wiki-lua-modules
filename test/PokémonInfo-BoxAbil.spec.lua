-- "Test cases" for PokémonInfo-BoxAbil

local b = require("PokémonInfo-BoxAbil").boxAbil

-- Standard Pokémon
print(b({ args = { "Staraptor" } }))

-- Multigen with "nessuna"
print(b({ args = { "Sableye" } }))

-- Multigen and alternative forms
print(b({ args = { "Gengar" } }))

-- Grouped alternative forms
print(b({ args = { "Pikachu" } }))

-- Special children
print(b({ args = { "Basculin" } }))
print(b({ args = { "Greninja" } }))
print(b({ args = { "Zygarde" } }))
print(b({ args = { "Rockruff" } }))
