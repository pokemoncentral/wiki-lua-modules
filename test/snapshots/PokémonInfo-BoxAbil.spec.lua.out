<div></div><div class="flex flex-row flex-wrap flex-items-center flex-main-stretch" style="padding: 0.2em;"><div class="width-xl-33 width-xs-50" style="padding: 0.2em;">[[Prepotenza|<span style="color:#000;">Prepotenza</span>]]<div class="small-text">Prima abilit&agrave;</div></div><div class="width-xl-33 width-xs-50" style="padding: 0.2em;">[[Temerarietà|<span style="color:#000;">Temerarietà</span>]]<div class="small-text">Abilit&agrave; speciale</div></div></div>
<div><div class="width-xl-100" style="padding-bottom: 0.2em;">'''Sableye'''</div></div><div class="flex flex-row flex-wrap flex-items-center flex-main-stretch" style="padding: 0.2em;"><div class="width-xl-33 width-xs-50" style="padding: 0.2em;">[[Sguardofermo|<span style="color:#000;">Sguardofermo</span>]]<div class="small-text">Prima abilit&agrave;</div></div><div class="width-xl-33 width-xs-50" style="padding: 0.2em;"><div>&mdash;<sup style="padding-left: 0.3em;">III</sup></div><div>[[Rallentatore|<span style="color:#000;">Rallentatore</span>]]<sup style="padding-left: 0.3em;">IV+</sup></div><div class="small-text">Seconda abilit&agrave;</div></div><div class="width-xl-33 width-xs-50" style="padding: 0.2em;">[[Burla|<span style="color:#000;">Burla</span>]]<div class="small-text">Abilit&agrave; speciale</div></div></div><div><div class="width-xl-100" style="padding-bottom: 0.2em;">'''MegaSableye'''</div></div><div class="flex flex-row flex-wrap flex-items-center flex-main-stretch" style="padding: 0.2em;"><div class="width-xl-33 width-xs-50" style="padding: 0.2em;">[[Magispecchio|<span style="color:#000;">Magispecchio</span>]]</div></div>
<div><div class="width-xl-100" style="padding-bottom: 0.2em;">'''Gengar'''</div></div><div class="flex flex-row flex-wrap flex-items-center flex-main-stretch" style="padding: 0.2em;"><div class="width-xl-33 width-xs-50" style="padding: 0.2em;"><div>[[Levitazione|<span style="color:#000;">Levitazione</span>]]<sup style="padding-left: 0.3em;">III-VI</sup></div><div>[[Corpofunesto|<span style="color:#000;">Corpofunesto</span>]]<sup style="padding-left: 0.3em;">VII+</sup></div></div></div><div><div class="width-xl-100" style="padding-bottom: 0.2em;">'''MegaGengar'''</div></div><div class="flex flex-row flex-wrap flex-items-center flex-main-stretch" style="padding: 0.2em;"><div class="width-xl-33 width-xs-50" style="padding: 0.2em;">[[Pedinombra|<span style="color:#000;">Pedinombra</span>]]</div></div><div><div class="width-xl-100" style="padding-bottom: 0.2em;">'''Gengar Gigamax'''</div></div><div class="flex flex-row flex-wrap flex-items-center flex-main-stretch" style="padding: 0.2em;"><div class="width-xl-33 width-xs-50" style="padding: 0.2em;">[[Corpofunesto|<span style="color:#000;">Corpofunesto</span>]]</div></div>
<div><div class="width-xl-100" style="padding-bottom: 0.2em;">'''Pikachu e Pikachu Gigamax'''</div></div><div class="flex flex-row flex-wrap flex-items-center flex-main-stretch" style="padding: 0.2em;"><div class="width-xl-33 width-xs-50" style="padding: 0.2em;">[[Statico|<span style="color:#000;">Statico</span>]]<div class="small-text">Prima abilit&agrave;</div></div><div class="width-xl-33 width-xs-50" style="padding: 0.2em;">[[Parafulmine|<span style="color:#000;">Parafulmine</span>]]<div class="small-text">Abilit&agrave; speciale</div></div></div><div><div class="width-xl-100" style="padding-bottom: 0.2em;">'''Pikachu Cosplay'''</div></div><div class="flex flex-row flex-wrap flex-items-center flex-main-stretch" style="padding: 0.2em;"><div class="width-xl-33 width-xs-50" style="padding: 0.2em;">[[Parafulmine|<span style="color:#000;">Parafulmine</span>]]</div></div>
<div><div class="width-xl-100" style="padding-bottom: 0.2em;">'''Forma Linearossa'''</div></div><div class="flex flex-row flex-wrap flex-items-center flex-main-stretch" style="padding: 0.2em;"><div class="width-xl-33 width-xs-50" style="padding: 0.2em;">[[Temerarietà|<span style="color:#000;">Temerarietà</span>]]<div class="small-text">Prima abilit&agrave;</div></div><div class="width-xl-33 width-xs-50" style="padding: 0.2em;">[[Adattabilità|<span style="color:#000;">Adattabilità</span>]]<div class="small-text">Seconda abilit&agrave;</div></div><div class="width-xl-33 width-xs-50" style="padding: 0.2em;">[[Rompiforma|<span style="color:#000;">Rompiforma</span>]]<div class="small-text">Abilit&agrave; speciale</div></div></div><div><div class="width-xl-100" style="padding-bottom: 0.2em;">'''Forma Lineablu'''</div></div><div class="flex flex-row flex-wrap flex-items-center flex-main-stretch" style="padding: 0.2em;"><div class="width-xl-33 width-xs-50" style="padding: 0.2em;">[[Testadura|<span style="color:#000;">Testadura</span>]]<div class="small-text">Prima abilit&agrave;</div></div><div class="width-xl-33 width-xs-50" style="padding: 0.2em;">[[Adattabilità|<span style="color:#000;">Adattabilità</span>]]<div class="small-text">Seconda abilit&agrave;</div></div><div class="width-xl-33 width-xs-50" style="padding: 0.2em;">[[Rompiforma|<span style="color:#000;">Rompiforma</span>]]<div class="small-text">Abilit&agrave; speciale</div></div></div><div><div class="width-xl-100" style="padding-bottom: 0.2em;">'''Forma Lineabianca'''</div></div><div class="flex flex-row flex-wrap flex-items-center flex-main-stretch" style="padding: 0.2em;"><div class="width-xl-33 width-xs-50" style="padding: 0.2em;">[[Paura|<span style="color:#000;">Paura</span>]]<div class="small-text">Prima abilit&agrave;</div></div><div class="width-xl-33 width-xs-50" style="padding: 0.2em;">[[Adattabilità|<span style="color:#000;">Adattabilità</span>]]<div class="small-text">Seconda abilit&agrave;</div></div><div class="width-xl-33 width-xs-50" style="padding: 0.2em;">[[Rompiforma|<span style="color:#000;">Rompiforma</span>]]<div class="small-text">Abilit&agrave; speciale</div></div></div>
<div><div class="width-xl-100" style="padding-bottom: 0.2em;">'''Greninja'''</div></div><div class="flex flex-row flex-wrap flex-items-center flex-main-stretch" style="padding: 0.2em;"><div class="width-xl-33 width-xs-50" style="padding: 0.2em;">[[Acquaiuto|<span style="color:#000;">Acquaiuto</span>]]<div class="small-text">Prima abilit&agrave;</div></div><div class="width-xl-33 width-xs-50" style="padding: 0.2em;">[[Mutatipo|<span style="color:#000;">Mutatipo</span>]]<div class="small-text">Abilit&agrave; speciale</div></div><div class="width-xl-33 width-xs-50" style="padding: 0.2em;">[[Morfosintonia|<span style="color:#000;">Morfosintonia</span>]]<div class="small-text">Abilit&agrave; evento</div></div></div><div><div class="width-xl-100" style="padding-bottom: 0.2em;">'''Forma Ash'''</div></div><div class="flex flex-row flex-wrap flex-items-center flex-main-stretch" style="padding: 0.2em;"><div class="width-xl-33 width-xs-50" style="padding: 0.2em;">[[Morfosintonia|<span style="color:#000;">Morfosintonia</span>]]</div></div>
<div><div class="width-xl-100" style="padding-bottom: 0.2em;">'''Forma 10% e Forma 50%'''</div></div><div class="flex flex-row flex-wrap flex-items-center flex-main-stretch" style="padding: 0.2em;"><div class="width-xl-33 width-xs-50" style="padding: 0.2em;">[[Frangiaura|<span style="color:#000;">Frangiaura</span>]]<div class="small-text">Prima abilit&agrave;</div></div><div class="width-xl-33 width-xs-50" style="padding: 0.2em;">[[Sciamefusione|<span style="color:#000;">Sciamefusione</span>]]<div class="small-text">Abilit&agrave; evento</div></div></div><div><div class="width-xl-100" style="padding-bottom: 0.2em;">'''Forma Perfetta'''</div></div><div class="flex flex-row flex-wrap flex-items-center flex-main-stretch" style="padding: 0.2em;"><div class="width-xl-33 width-xs-50" style="padding: 0.2em;">[[Sciamefusione|<span style="color:#000;">Sciamefusione</span>]]</div></div>
<div></div><div class="flex flex-row flex-wrap flex-items-center flex-main-stretch" style="padding: 0.2em;"><div class="width-xl-33 width-xs-50" style="padding: 0.2em;">[[Sguardofermo|<span style="color:#000;">Sguardofermo</span>]]<div class="small-text">Prima abilit&agrave;</div></div><div class="width-xl-33 width-xs-50" style="padding: 0.2em;">[[Spiritovivo|<span style="color:#000;">Spiritovivo</span>]]<div class="small-text">Seconda abilit&agrave;</div></div><div class="width-xl-33 width-xs-50" style="padding: 0.2em;">[[Cuordeciso|<span style="color:#000;">Cuordeciso</span>]]<div class="small-text">Abilit&agrave; speciale</div></div><div class="width-xl-33 width-xs-50" style="padding: 0.2em;">[[Mente Locale|<span style="color:#000;">Mente Locale</span>]]<div class="small-text">Abilit&agrave; evento</div></div></div>
